reflex/components/tags/iter_tag.py

```diff
@@ -30,9 +30,11 @@ def get_index_var() -> Var:
         Returns:
             The index var.
         """
-        index = Var.create(INDEX_VAR, is_local=False)
-        assert index is not None
-        return index
+        return BaseVar(
+            name=INDEX_VAR,
+            type_=int,
+            is_local=True,
+        )
 
     @staticmethod
     def get_index_var_arg() -> Var:
```

Comment:
Shouldn't we change the type annotation if we are now returning BaseVar instead of Var?